--- a/lxc-fedora	2015-06-21 07:33:20.288395841 +0100
+++ b/lxc-fedora	2015-06-21 07:33:59.004394032 +0100
@@ -687,7 +687,7 @@
     echo "Stage 1 creation complete.  Building stage 2 Installation Bootstrap"
 
     mount -o bind ../bootstrap run/install
-    rsync -av ${mirrorurl}/fedora/linux/releases/20/Fedora/$basearch/os/Packages/f/fedora-release-20* .
+    rsync -av ${mirrorurl}/fedora/linux/releases/22/Fedora/$basearch/os/Packages/f/fedora-release-22* .
 
     # The --nodeps is STUPID but F15 had a bogus dependency on RawHide?!?!
     chroot . rpm --root /run/install --nodeps -ivh fedora-release-*
@@ -695,7 +695,7 @@
     # yum will take $basearch from host, so force the arch we want
     sed -i "s|\$basearch|$basearch|" ./run/install/etc/yum.repos.d/*
 
-    chroot . yum -y --nogpgcheck --installroot /run/install install python rpm yum
+    chroot . dnf -y --nogpgcheck --installroot /run/install install python rpm dnf
 
     umount run/install
     umount proc
@@ -809,7 +809,7 @@
     BOOTSTRAP_CHROOT=
     BOOTSTRAP_DIR=
 
-    PKG_LIST="yum initscripts passwd rsyslog vim-minimal openssh-server openssh-clients dhclient chkconfig rootfiles policycoreutils fedora-release"
+    PKG_LIST="dnf yum initscripts passwd rsyslog vim-minimal openssh-server openssh-clients dhclient chkconfig rootfiles policycoreutils fedora-release"
     MIRRORLIST_URL="http://mirrors.fedoraproject.org/mirrorlist?repo=fedora-$release&arch=$basearch"
 
     if [[ ${release} -lt 17 ]]
@@ -1016,7 +1016,7 @@
     mount -t proc proc ${cache}/rootfs/proc
     # Always make sure /etc/resolv.conf is up to date in the target!
     cp /etc/resolv.conf ${cache}/rootfs/etc/
-    chroot ${cache}/rootfs yum -y update
+    chroot ${cache}/rootfs dnf -y update
     umount ${cache}/rootfs/proc
     umount  ${cache}/rootfs/dev
 }
