--- a/lxc-fedora	2015-02-24 07:33:20.288395841 +0100
+++ b/lxc-fedora	2015-02-24 07:33:59.004394032 +0100
@@ -562,7 +562,7 @@
     if [[ ! -f ../LiveOS/squashfs.img ]]
     then
         echo "
-Downloading stage 0 LiveOS squashfs file system from mirrors.kernel.org...
+Downloading stage 0 LiveOS squashfs file system from Fedora mirrors...
 Have a beer or a cup of coffee.  This will take a bit (~300MB).
 "
         sleep 3 # let him read it...
@@ -570,7 +570,7 @@
         # Right now, we are using Fedora 20 for the inial bootstrap.
         # We could make this the "current" Fedora rev (F > 15).

-        rsync -av mirrors.kernel.org::fedora/releases/20/Fedora/$basearch/os/LiveOS .
+        rsync -av ${mirrorurl}/fedora/linux/releases/20/Fedora/$basearch/os/LiveOS .

         if [[ 0 == $? ]]
         then
@@ -647,8 +647,8 @@
     # of this LiveOS image we're camped out on.  This is the beginning
     # of the butt ugly hack.  Look close or you may missing it...

-    rsync -av mirrors.kernel.org::fedora/releases/20/Fedora/$basearch/os/Packages/r/rpm-[0-9]* \
-        mirrors.kernel.org::fedora/releases/20/Fedora/$basearch/os/Packages/y/yum-[0-9]* .
+    rsync -av ${mirrorurl}/fedora/linux/releases/20/Fedora/$basearch/os/Packages/r/rpm-[0-9]* \
+        ${mirrorurl}/fedora/linux/releases/20/Fedora/$basearch/os/Packages/y/yum-[0-9]* .

     # And here it is...
     # The --nodeps is STUPID but F15 had a bogus dependency on RawHide?!?!
@@ -680,7 +680,7 @@
     echo "Stage 1 creation complete.  Building stage 2 Installation Bootstrap"

     mount -o bind ../bootstrap run/install
-    rsync -av mirrors.kernel.org::fedora/releases/20/Fedora/$basearch/os/Packages/f/fedora-release-20* .
+    rsync -av ${mirrorurl}/fedora/linux/releases/20/Fedora/$basearch/os/Packages/f/fedora-release-20* .

     # The --nodeps is STUPID but F15 had a bogus dependency on RawHide?!?!
     chroot . rpm --root /run/install --nodeps -ivh fedora-release-*
@@ -813,6 +813,13 @@
         PKG_LIST="${PKG_LIST} db4-utils"
     fi

+    if [[ ${release} -ge 21 ]]
+    then
+        # Since Fedora 21, a separate fedora-repos package is needed.
+        # Before, the information was conained in fedora-release.
+        PKG_LIST="${PKG_LIST} fedora-repos"
+    fi
+
     DOWNLOAD_OK=no
 
     # We're splitting the old loop into two loops plus a directory retrival.
@@ -1191,6 +1198,12 @@
 *) ;;
 esac

+mirrorurl="archives.fedoraproject.org::fedora-archive"
+case "$basearch" in
+ppc64|s390x) mirrorurl="archives.fedoraproject.org::fedora-secondary" ;;
+*) ;;
+esac
+
 # Somebody wants to specify an arch.  This is very limited case.
 #       i386/i586/i686 on i386/x86_64
 #           - or -

